test/IntegrationTests/Log4NetBridgeTests.cs

```diff
@@ -0,0 +1,146 @@
+        using var collector = new MockLogsCollector(Output);
+        SetExporter(collector);
+
+        // Logged in scope of an activity
+        collector.Expect(logRecord =>
+            VerifyBody(logRecord, "Hello, world!") &&
+            VerifyTraceContext(logRecord) &&
+            logRecord is { SeverityText: "INFO", SeverityNumber: SeverityNumber.Info } &&
+            logRecord.Attributes.Count == 0);
+
+        // Logged with exception
+        collector.Expect(logRecord =>
+            VerifyBody(logRecord, "Exception occured") &&
+            VerifyTraceContext(logRecord) &&
+            logRecord is { SeverityText: "ERROR", SeverityNumber: SeverityNumber.Error } &&
+            VerifyExceptionAttributes(logRecord) &&
+            logRecord.Attributes.Count == 3);
+
+        EnableBytecodeInstrumentation();
+
+        var (standardOutput, _, _) = RunTestApplication(new()
+        {
+            PackageVersion = packageVersion,
+            Arguments = "--api log4net"
+        });
+
+        AssertStandardOutputExpectations(standardOutput);
+
+        collector.AssertExpectations();
+    }
+
+#if !NETFRAMEWORK
+    [Theory]
+    [Trait("Category", "EndToEnd")]
+    [MemberData(nameof(LibraryVersion.log4net), MemberType = typeof(LibraryVersion))]
+    public void SubmitLogs_ThroughILoggerBridge_WhenLog4NetIsUsedAsILoggerAppenderForLogging(string packageVersion)
+    {
+        using var collector = new MockLogsCollector(Output);
+        SetExporter(collector);
+
+        // Logged in scope of an activity
+        collector.Expect(logRecord =>
+            VerifyBody(logRecord, "Hello, {0}!") &&
+            VerifyTraceContext(logRecord) &&
+            logRecord is { SeverityText: "Information", SeverityNumber: SeverityNumber.Info } &&
+            // ILogger bridge adds original format when using structured logging.
+            logRecord.Attributes.Count == 1);
+
+        // Logged with exception
+        collector.Expect(logRecord =>
+            VerifyBody(logRecord, "Exception occured") &&
+            VerifyTraceContext(logRecord) &&
+            // OtlpLogExporter adds exception related attributes (ConsoleExporter doesn't show them)
+            logRecord is { SeverityText: "Error", SeverityNumber: SeverityNumber.Error } &&
+            VerifyExceptionAttributes(logRecord) &&
+            logRecord.Attributes.Count == 3);
+
+        EnableBytecodeInstrumentation();
+
+        var (standardOutput, _, _) = RunTestApplication(new()
+        {
+            PackageVersion = packageVersion,
+            Arguments = "--api ILogger"
+        });
+
+        AssertStandardOutputExpectations(standardOutput);
+
+        collector.AssertExpectations();
+    }
+
+    [Theory]
+    [Trait("Category", "EndToEnd")]
+    [MemberData(nameof(LibraryVersion.log4net), MemberType = typeof(LibraryVersion))]
+    public async Task SubmitLogs_ThroughILoggerBridge_WhenLog4NetIsUsedAsILoggerAppenderForLogging_WithoutDuplicates(string packageVersion)
```

Comment:
Is this testing anything different than the above ILogger one?

Reply:
Test above won't fail if there are additional log records, which were not expected.